clang/lib/DPCT/PatternRewriter.cpp

```diff
@@ -388,6 +393,33 @@ static bool isIdentifiedChar(char Char) {
   return false;
 }
 
+static void
+updateExtentionName(const std::string &Input, size_t Next,
+                    std::unordered_map<std::string, std::string> &Bindings) {
+  if (Input.compare(Next, strlen(".cpp"), ".cpp") == 0) {
+    size_t Pos = Next - 1;
+    for (; Pos > 0 && isIdentifiedChar(Input[Pos]); Pos--) {
+    }
+    Pos = Pos == 0 ? 0 : Pos + 1;
+    std::string FileName = Input.substr(Pos, Next + strlen(".cpp") - 1 - Pos);
+
+    std::string SyclFileName;
+    rewriteFileName(SyclFileName, FileName);
+    bool HasCudaSyntax = false;
+    for (const auto &File : MainSrcFilesHasCudaSyntex) {
+      if (File.find(FileName) != std::string::npos) {
+        HasCudaSyntax = true;
+      }
+    }
+    if (HasCudaSyntax)
+      Bindings["rewrite_extention_name"] = "cpp.dp.cpp";
```

Comment:
Please fix the hardcode name in following PR in L415.

Reply:
Will update this in a separate PR.